user_external/lib/imap.php

```diff
@@ -49,6 +49,32 @@ public function checkPassword($uid, $password) {
 			imap_close($mbox);
 			$uid = mb_strtolower($uid);
 			$this->storeUser($uid);
+			
+			if (preg_match('/^([^@]+)@(.+)\.[^.]+$/', $uid, $matches)) { 
```

Comment:
I checked your suggestion on my system.
It worked fine.
Now I am not expert enough to change my patch.
The new lines would be:

  if (\OC::$server->getMailer()->validateMailAddress($uid)) {
                          preg_match('/^([^@]+)@(.+)\.[^.]+$/', $uid, $matches);